new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodsRestart.java

```diff
@@ -0,0 +1,370 @@
+
+    // install and verify operator
+    installAndVerifyOperator(opNamespace, domainNamespace);
+
+    // create a basic model in image domain
+    createAndVerifyMiiDomain();
+  }
+
+  /**
+   * Add/Modify server pod resources by patching the domain custom resource.
+   * Verify all pods were restarted and back to be ready.
+   * The resources tested: resources: limits: cpu: "1", resources: requests: cpu: "0.5"
+   * Test fails if any server pod was not restarted and back to be ready or the compute resources in the patched
+   * domain custom resource do not match the values we planed to add or modify.
+   */
+  @Test
+  @DisplayName("Verify server pods were restarted by changing the resources")
+  public void testServerPodsRestartByChangingResource() {
+
+    // get the original domain resource before update
+    Domain domain1 = assertDoesNotThrow(() -> getDomainCustomResource(domainUid, domainNamespace),
+        String.format("getDomainCustomResource failed with ApiException when tried to get domain %s in namespace %s",
+            domainUid, domainNamespace));
+
+    assertNotNull(domain1, "domain1 is null");
+    assertNotNull(domain1.getSpec(), "domain1/spec is null");
+    assertNotNull(domain1.getSpec().getServerPod(), "domain1/spec/serverPod is null");
+    assertNotNull(domain1.getSpec().getServerPod().getResources(), "domain1/spec/serverPod/resources is null");
+
+    // get the current server pod compute resource limit
+    Map<String, Quantity> limits = domain1.getSpec().getServerPod().getResources().getLimits();
+    assertNotNull(limits, "domain1/spec/serverPod/resources/limits is null");
+
+    // print out current server pod compute resource limits
+    logger.info("Current value for server pod compute resource limits:");
+    limits.forEach((key, value) -> logger.info(key + ": " + value.toString()));
+
+    // get the current server pod compute resource requests
+    Map<String, Quantity> requests = domain1.getSpec().getServerPod().getResources().getRequests();
+    assertNotNull(requests, "domain1/spec/serverPod/resources/requests is null");
+
+    // print out current server pod compute resource requests
+    logger.info("Current value for server pod compute resource requests:");
+    requests.forEach((key, value) -> logger.info(key + ": " + value.toString()));
+
+    // get the admin server pod original creation timestamp
+    logger.info("Getting admin server pod original creation timestamp");
+    String adminPodOriginalTimestamp =
+        assertDoesNotThrow(() -> getPodCreationTimestamp(domainNamespace, "", adminServerPodName),
+            String.format("getPodCreationTimestamp failed with ApiException for pod %s in namespace %s",
+                adminServerPodName, domainNamespace));
+
+    // get the managed server pods original creation timestamps
+    logger.info("Getting managed server pods original creation timestamps");
+    List<String> managedServerPodOriginalTimestampList = new ArrayList<>();
```

Comment:
@sankarpn, @anpanigr, @xiancao... Including all of the pending PR's, it seems like we have three different ways to assert that a pod has been restarted. Let's see if we can come up with a consistent pattern.

Reply:
Can you list the other ways to assert that a pod has been restarted? In TestAssertions.java, there is only one assertion related to pod restart.

Reply:
from yesterday's meeting discussion, the cluster server will be rolling restarted after the resources change. I will use Sankar's assertion method once it is ready.

Reply:
updated the code using verifyRollingRestartOccurred()